spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -251,17 +251,34 @@ class Analyzer(tableUtils: TableUtils,
 
   def analyzeJoin(joinConf: api.Join,
                   enableHitter: Boolean = false,
+                  validateTablePermission: Boolean = true,
                   validationAssert: Boolean = false): (Map[String, DataType], ListBuffer[AggregationMetadata]) = {
     val name = "joins/" + joinConf.metaData.name
     logger.info(s"""|Running join analysis for $name ...""".stripMargin)
     // run SQL environment setups such as UDFs and JARs
     joinConf.setups.foreach(tableUtils.sql)
 
-    val leftDf = JoinUtils.leftDf(joinConf, range, tableUtils, allowEmpty = true).get
-    val analysis = if (enableHitter) analyze(leftDf, joinConf.leftKeyCols, joinConf.left.table) else ""
-    val leftSchema: Map[String, DataType] =
-      leftDf.schema.fields.map(field => (field.name, SparkConversions.toChrononType(field.name, field.dataType))).toMap
-
+    val (analysis, leftSchema) = if (enableHitter) {
```

Comment:
One more suggestion to simplify, you can return `leftDf` here instead of `leftSchema`. Then you don't need to duplicate this logic:

```
        .schema
        .fields
        .map(field => (field.name, SparkConversions.toChrononType(field.name, field.dataType)))
        .toMap
```